File: src/_1ProgrammingBasicsWithJavaMay2023/_03ConditionalStatementsAdvanced/_2Exercise/_08OnTimeForTheExam.java
package _1ProgrammingBasicsWithJavaMay2023._03ConditionalStatementsAdvanced._2Exercise;

import java.util.Scanner;

public class _08OnTimeForTheExam {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int hoursOfExam = Integer.parseInt(scanner.nextLine());
        int minutesOfExam = Integer.parseInt(scanner.nextLine());
        int hoursOfArrival = Integer.parseInt(scanner.nextLine());
        int minutesOfArrival = Integer.parseInt(scanner.nextLine());
        int timeForExam = (hoursOfExam * 60) + minutesOfExam;
        int timeForArrival = (hoursOfArrival * 60) + minutesOfArrival;
        int diff = Math.abs(timeForArrival - timeForExam);
        if (timeForArrival > timeForExam) {
            System.out.println("Late");
            if (diff >= 60) {
                int hours = diff / 60;
                int minutes = diff % 60;
                System.out.printf("%d:%02d hours after the start", hours, minutes);
            } else {
                System.out.printf("%d minutes after the start", diff);
            }
        } else if (timeForArrival == timeForExam || diff <= 30) {
            System.out.println("On Time");
            if (diff >= 1 && diff <= 30) {
                System.out.printf("%d minutes before the start", diff);
            }
        } else {
            System.out.println("Early");
            if (diff >= 60) {
                int hours = diff / 60;
                int minutes = diff % 60;
                System.out.printf("%d:%02d hours before the start", hours, minutes);
            } else {
                System.out.printf("%d minutes before the start", diff);
            }
        }
    }
}

File: src/_1ProgrammingBasicsWithJavaScriptNovember2024/_3ConditionalStatementsAdvanced/_2Exercise/_08OnTimeForTheExam.js
function onTimeForTheExam(hourOfExam, minutesOfExam, hourOfStudentCome, minutesOfStudentCome) {
    const examMinutesStart = hourOfExam * 60 + minutesOfExam;
    const studentMinutesCome = hourOfStudentCome * 60 + minutesOfStudentCome;

    const timeDifference = examMinutesStart - studentMinutesCome;

    const hours = Math.floor(Math.abs(timeDifference) / 60);
    const minutes = Math.abs(timeDifference) % 60;

    let timing = "";

    if (timeDifference > 30) {
        console.log(`Early`);
        timing = "before the start";
    } else if (timeDifference >= 0) {
        console.log(`On time`);
        timing = "before the start";
    } else {
        console.log(`Late`);
        timing = "after the start";
    }

    if (hours > 0) {
        console.log(`${hours}:${minutes.toString().padStart(2, '0')} hours ${timing}`);
    } else if (minutes > 0) {
        console.log(`${minutes} minutes ${timing}`);
    }
}